### PR TITLE
bugfix for stress loss

### LINKED ISSUE #437: Stress / num_atoms in loss?
Why is stress divided by num_atoms here? Maybe a bug?

https://github.com/ACEsuit/mace/blob/dee204f1f9d587f28fd792fdad1f45039ef71e94/mace/modules/loss.py#L30C1-L39C12
```
def weighted_mean_squared_stress(ref: Batch, pred: TensorDict) -> torch.Tensor:
    # energy: [n_graphs, ]
    configs_weight = ref.weight.view(-1, 1, 1)  # [n_graphs, ]
    configs_stress_weight = ref.stress_weight.view(-1, 1, 1)  # [n_graphs, ]
    num_atoms = (ref.ptr[1:] - ref.ptr[:-1]).view(-1, 1, 1)  # [n_graphs,]
    return torch.mean(
        configs_weight
        * configs_stress_weight
        * torch.square((ref["stress"] - pred["stress"]) / num_atoms)
    )  # []
```

The Huber loss looks like it doesn't have the division by num_atoms.

```
class WeightedHuberEnergyForcesStressLoss(torch.nn.Module):
    def __init__(
        self, energy_weight=1.0, forces_weight=1.0, stress_weight=1.0, huber_delta=0.01
    ) -> None:
        super().__init__()
        self.huber_loss = torch.nn.HuberLoss(reduction="mean", delta=huber_delta)
        self.register_buffer(
            "energy_weight",
            torch.tensor(energy_weight, dtype=torch.get_default_dtype()),
        )
        self.register_buffer(
            "forces_weight",
            torch.tensor(forces_weight, dtype=torch.get_default_dtype()),
        )
        self.register_buffer(
            "stress_weight",
            torch.tensor(stress_weight, dtype=torch.get_default_dtype()),
        )

    def forward(self, ref: Batch, pred: TensorDict) -> torch.Tensor:
        num_atoms = ref.ptr[1:] - ref.ptr[:-1]
        return (
            self.energy_weight
            * self.huber_loss(ref["energy"] / num_atoms, pred["energy"] / num_atoms)
            + self.forces_weight * self.huber_loss(ref["forces"], pred["forces"])
            + self.stress_weight * self.huber_loss(ref["stress"], pred["stress"])
```

### PR DESCRIPTION
**Pull request by JPDarby**
Removed division of stress by n_atoms in loss to resolve #437

### CODE REVIEW

**Comment by bernstei** [comment]
I noticed that the train.py error report says `stress_per_atom`, which makes me worry - stress is intensive, it shouldn't be "per atom" the way I usually use that phrase. As I poke around the code I see `delta_stress` and `delta_stress_per_atom`, e.g. https://github.com/ACEsuit/mace/blob/dee204f1f9d587f28fd792fdad1f45039ef71e94/mace/tools/train.py#L422

Can someone (@ilyes319 ?) confirm that this kind of bug isn't present elsewhere in the code?

I'm hoping that those variables mean "total stress" and "atomic stress", rather than `per_atom` meaning divided by n_atoms. However, even if it is, depending on what exactly the atomic stress is (i.e. dE_site / dVol_total, or dE_site / dVol_site (= V_total / n_atoms?), it may or may not be what people are expecting.
